dev/.rat-excludes

```diff
@@ -9,6 +9,7 @@ build
 .rat-excludes
 LICENSE
 NOTICE
+revapi.yml
```

Comment:
YAML can have comments: https://www.tutorialspoint.com/yaml/yaml_comments.htm

I think this should just add the ASF header instead of updating `.rat-excludes`.

Reply:
The reason I didn’t add it is because this file is autogenerated.

I can try adding it, forcing an update to the file, and then seeing if it handles the header well.

Reply:
Unfortunately, it will have to go in the `.rat-excludes` file.

I added the header, then ran `./gradlew iceberg-api:revapiAcceptAllBreaks --justification "Testing YAML comment on top of file"` to cause an update to the file, and the resulting generated file removed the header comment.

So unless we want to manually re-enter the license header every time we make a chance to the file, it needs to be excluded.

Reply:
Here's part of the diff after forcing a chance in the file:

```
diff --git a/.palantir/revapi.yml b/.palantir/revapi.yml
index 03fc8d427..f310c4923 100644
--- a/.palantir/revapi.yml
+++ b/.palantir/revapi.yml
@@ -1,21 +1,85 @@
-#
-# Licensed to the Apache Software Foundation (ASF) under one
-# or more contributor license agreements.  See the NOTICE file
-# distributed with this work for additional information
-# regarding copyright ownership.  The ASF licenses this file
-# to you under the Apache License, Version 2.0 (the
-# "License"); you may not use this file except in compliance
-# with the License.  You may obtain a copy of the License at
-#
-#   http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing,
-# software distributed under the License is distributed on an
-# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
-# KIND, either express or implied.  See the License for the
-# specific language governing permissions and limitations
-# under the License.
-#
-
 versionOverrides:
   org.apache.iceberg:iceberg-api:release-base-0.13.0: "0.13.1"
+acceptedBreaks:
+  release-base-0.13.0:
+    org.apache.iceberg:iceberg-api:
+    - code: "java.class.defaultSerializationChanged"
+      old: "class org.apache.iceberg.PartitionSpec"
+      new: "class org.apache.iceberg.PartitionSpec"
+      justification: "Testing YAML comment on top of file"
+    - code: "java.class.defaultSerializationChanged"
+      old: "class org.apache.iceberg.Schema"
+      new: "class org.apache.iceberg.Schema"
+      justification: "Testing YAML comment on top of file"
```

Reply:
If it's auto-generated, it should be fine.